apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -0,0 +1,749 @@
+        representatives[msg.sender] = _representative;
+        emit ChangeRepresentative(msg.sender, _representative);
+    }
+
+    /**
+    * @notice Create a new vote about "`_metadata`"
+    * @param _executionScript EVM script to be executed on approval
+    * @param _metadata Vote metadata
+    * @return voteId id for newly created vote
+    */
+    function newVote(bytes _executionScript, string _metadata) external auth(CREATE_VOTES_ROLE) returns (uint256 voteId) {
+        return _newVote(_executionScript, _metadata);
+    }
+
+    /**
+    * @notice Vote `_supports ? 'yes' : 'no'` in vote #`_voteId`
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote()`, which requires initialization
+    * @param _voteId Id for vote
+    * @param _supports Whether voter supports the vote
+    */
+    function vote(uint256 _voteId, bool _supports) external voteExists(_voteId) {
+        require(_canVote(votes[_voteId], msg.sender), ERROR_CANNOT_VOTE);
+        _vote(_voteId, _supports, msg.sender);
+    }
+
+    /**
+    * @notice Vote `_supports ? 'yes' : 'no'` in vote #`_voteId` on behalf of a set of voters
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote()`, which requires initialization
+    * @param _voteId Id for vote
+    * @param _supports Whether the representative supports the vote
+    * @param _voters Addresses of the voters voting on behalf of
+    */
+    function voteOnBehalfOf(uint256 _voteId, bool _supports, address[] _voters) external voteExists(_voteId) {
+        require(_voters.length <= MAX_VOTES_DELEGATION_SET_LENGTH, ERROR_DELEGATES_EXCEEDS_MAX_LEN);
+        _voteOnBehalfOf(_voteId, _supports, _voters);
+    }
+
+    /**
+    * @notice Execute vote #`_voteId`
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote()`, which requires initialization
+    * @param _voteId Id for vote
+    */
+    function executeVote(uint256 _voteId) external voteExists(_voteId) {
+        Vote storage vote_ = votes[_voteId];
+        require(_canExecute(vote_), ERROR_CANNOT_EXECUTE);
+
+        vote_.status = VoteStatus.Executed;
+        _closeAgreementAction(vote_.actionId);
+
+        bytes memory input = new bytes(0); // TODO: Consider input for voting scripts
+        address[] memory blacklist = new address[](1);
+        blacklist[0] = address(_getAgreement());
+        runScript(vote_.executionScript, input, blacklist);
+        emit ExecuteVote(_voteId);
+    }
+
+    // Disputable getter fns
+
+    /**
+    * @dev Tells whether a vote can be challenged or not
+    * @return True if the given vote can be challenged, false otherwise
+    */
+    function canChallenge(uint256 _voteId) external view returns (bool) {
+        return _voteExists(_voteId) && _isVoteOpen(votes[_voteId]);
```

Comment:
Wondering if we should lift the `_voteExists()` into the modifier so it reverts instead like the other getters?